lib/doorkeeper/openid_connect/oauth/pre_authorization.rb

```diff
@@ -8,6 +8,22 @@ def initialize(server, attrs = {})
           super
           @nonce = attrs[:nonce]
         end
+
+        # This method will be updated when doorkeeper move to version > 5.2.2
+        # TODO: delete this method and refactor response_on_fragment? method (below) when doorkeeper gem version constrains is > 5.2.2
```

Comment:
setting `response_on_fragment` if `response_type` is one of following: `token`, `id_token`, `id_token token`

**TODO** explain:
with this [doorkeeper pull request](https://github.com/doorkeeper-gem/doorkeeper/pulls?q=is%3Apr+is%3Aclosed), `error_response` is refactor to make doorkeeper-oidc easier to override this behavior.

Reply:
Thanks for this! :+1: